python/lsst/pipe/analysis/coaddAnalysis.py

```diff
@@ -477,16 +491,29 @@ def readSrcMatches(self, dataRefList, dataset, hscRun=None, wcs=None, aliasDictL
             # Set some aliases for differing schema naming conventions
             if aliasDictList:
                 catalog = setAliasMaps(catalog, aliasDictList)
-            if dataset != "deepCoadd_meas" and any(ss not in catalog.schema
-                                                   for ss in self.config.columnsToCopy):
-                unforced = dataRef.get("deepCoadd_meas", immediate=True,
+            if dataset != (self.config.coaddName + "Coadd_meas" and
+                           any(ss not in catalog.schema for ss in list(self.config.columnsToCopyFromMeas) +
+                               list(self.config.columnsToCopyFromRef))):
+                # copy over some fields from _ref and _meas catalogs to _forced_src catalog
+                refCat = dataRef.get(self.config.coaddName + "Coadd_ref", immediate=True,
+                                     flags=afwTable.SOURCE_IO_NO_FOOTPRINTS)
+                unforced = dataRef.get(self.config.coaddName + "Coadd_meas", immediate=True,
                                        flags=afwTable.SOURCE_IO_NO_FOOTPRINTS)
-                # copy over some fields from unforced to forced catalog
+                if len(catalog) != len(refCat):
+                    raise RuntimeError(("Lengths of forced (N = {0:d}) and ref (N = {0:d}) cats don't match").
+                                       format(len(catalog), len(refCat)))
+                refColList = [s for s in refCat.schema.getNames() if
+                              s.startswith(tuple(self.config.columnsToCopyFromRef))]
+                catalog = addColumnsToSchema(refCat, catalog,
+                                             [col for col in refColList if col not in catalog.schema and
+                                             col in refCat.schema and not
+                                             (hscRun and col == "slot_Centroid_flag")])
```

Comment:
This seems to be the same list as earlier and could be reused. It would also be cleaner to reuse refColList rather than redefining it.

Reply:
This is in a different function and, not only are they both guaranteed to be run (e.g. if `plotMatchesOnly == True`), but I'd like to leave the possibility of allowing them to be different for the `match` catalog case, so have left as is (except to update for your preferred `columnsToCopy` predefinition as above).